### ISSUE TITLE
tests failing for edge functions

### ISSUE THREAD

**Issue by robitalec** [ropensci/spatsoc#23]
Three warnings/errors for `edge_dist` and `edge_nn`:

`longer object length is not a multiple of shorter object length`

`Different branches of j expression produced different auto-named columns: [NA!=distance]; using the most "last" names`

When "returnDist" is TRUE, this error is added:
`Internal error: jvnames is length 2 but ans is 4 and bynames is 1`

**Comment by robitalec** [comment]
This was solved by assigning, within the j `{}`, the output list. 

Eg. for `edge_nn`, from:

```r
    if (returnDist) {
      w <- wm + (length(wm) * (as.numeric(names(wm)) - 1))
      list(ID = .SD[[1]][as.numeric(names(wm))],
           NN = .SD[[1]][wm],
           distance = distMatrix[w])
    } else {
      list(ID = .SD[[1]][as.numeric(names(wm))],
           NN = .SD[[1]][wm])
    }
```

to:
```r
    if (returnDist) {
      w <- wm + (length(wm) * (as.numeric(names(wm)) - 1))
      l <- list(ID = .SD[[1]][as.numeric(names(wm))],
                NN = .SD[[1]][wm],
                distance = distMatrix[w])
    } else {
      l <- list(ID = .SD[[1]][as.numeric(names(wm))],
                NN = .SD[[1]][wm])
    }
    l
```

The if/else block here depends on a variable ("returnDist") passed to the function, that is set outside of the data.table chunk and doesn't change within it.

**Comment by robitalec** [comment]
Warnings+error present with `data.table` v.12.8 and not present with `data.table` v1.12.6.

Possibly related to https://github.com/Rdatatable/data.table/pull/3802

**Comment by robitalec** [comment]
Closed by #25
